### PR TITLE
Add Python 3.8 Support in pip

### PR DESCRIPTION
**Pull request by zhreshold**
*Issue #, if available:*
#120 

*Description of changes:*
Test python 3.8 in ci pipeline

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mseeger** [comment]
Could somebody comment here on what the issues are that users have with Python 3.8?
I myself currently have a situation where I have to use Python 3.8.5, so I'd be interested, so I can choose a workaround.

**Comment by zhreshold** [comment]
@mseeger here are some of the issues https://github.com/awslabs/autogluon/issues?q=is%3Aissue+python+3.8+ and I am sure it's not the full list

**Comment by mseeger** [comment]
So it is ConfigSpace. I'll try myself to get it working (I am working on EC2, not Windows), and can let you know.

**Comment by mseeger** [comment]
Sounds like it's also about Windows ...

**Comment by szha** [comment]
Job PR-955-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-955/3/index.html

File: core/src/autogluon/core/_setup_utils.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '..', '..')
 )
 
-PYTHON_REQUIRES = '>=3.6, <3.8'
+PYTHON_REQUIRES = '>=3.6, <3.9'
```
Comment (Innixma) [inline core/src/autogluon/core/_setup_utils.py:13]: Unrelated to this PR, but I see a warning printed on importing autogluon.vision that can probably be fixed: http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-955/3/tutorials/image_prediction/beginner.html

Comment (Innixma) [inline core/src/autogluon/core/_setup_utils.py:13]: Given this appears to work and has passed CI, should we merge this PR? Perhaps the ConfigSpace update fixed all the issues people were having with 3.8.

Comment (Innixma) [reply core/src/autogluon/core/_setup_utils.py:13]: @jwmueller @sxjscience @gradientsky

**Review by sxjscience — APPROVED** [review]
LGTM

**Comment by Innixma** [comment]
If we do merge, please rename PR so commit is representative of contribution (Added 3.8 support)

File: core/src/autogluon/core/_setup_utils.py (hunk 1 above)
Comment (jwmueller) [reply core/src/autogluon/core/_setup_utils.py:13]: I agree with updating the `PYTHON_REQUIRES` to include 3.8. Not sure the CI (build.yml) should be 3.8 or 3.7 though, may still want to keep it at 3.7 if that's where most users are.

**Comment by Innixma** [comment]
@zhreshold Can you keep CI to 3.7 for now? I think we can merge this PR to enable 3.8 for pip installs though.

**Review by Innixma — APPROVED** [review]

**Comment by szha** [comment]
Job PR-955-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-955/4/index.html

**Comment by szha** [comment]
Job PR-955-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-955/5/index.html

**Comment by szha** [comment]
Job PR-955-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-955/6/index.html

**Pull request merged**